invenio_circulation/api.py

```diff
@@ -92,6 +95,41 @@ def __init__(self, data, model=None):
             finalize_event='save',
         )
 
+    @classmethod
+    def item_available(cls, item_pid):
+        """."""
+        item_available = current_app.config.get('CIRCULATION_POLICIES').get(
+            'item_available'
+        )
+        if not item_available(item_pid):
+            return False
+        for loan in cls.get_loans(item_pid):
+            if loan.state != 'ITEM_RETURNED':
+                return False
+        return True
+
+    @classmethod
+    def get_loans(cls, item_pid):
+        """."""
+        search = LoansSearch()
+        search = search.filter('term', item_pid=item_pid).source(
+            includes='loan_pid'
+        )
+        for result in search.scan():
+            if result.loan_pid:
+                yield cls.get_record_by_pid(result.loan_pid)
+
+    @classmethod
+    def get_record_by_pid(cls, pid, with_deleted=False):
+        """Get ils record by pid value."""
+        resolver = Resolver(
```

Comment:
I don't think you want the `Resolver` here. You can ask directly PersistentIdentifier given the pid.

Reply:
Right